### PR TITLE
issue #75 alternative fix

### PR DESCRIPTION
**Pull request by donequis**

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 12 issues
* ![MAJOR][MAJOR] 7 major
* ![INFO][INFO] 5 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [TrieImpl.java#L535](https://github.com/rsksmart/rskj/blob/b4eff67aba79fb7a79fe3512d192477fb402ac89/rskj-core/src/main/java/co/rsk/trie/TrieImpl.java#L535): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [TrieImpl.java#L538](https://github.com/rsksmart/rskj/blob/b4eff67aba79fb7a79fe3512d192477fb402ac89/rskj-core/src/main/java/co/rsk/trie/TrieImpl.java#L538): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [TrieImpl.java#L548](https://github.com/rsksmart/rskj/blob/b4eff67aba79fb7a79fe3512d192477fb402ac89/rskj-core/src/main/java/co/rsk/trie/TrieImpl.java#L548): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [TrieImpl.java#L623](https://github.com/rsksmart/rskj/blob/b4eff67aba79fb7a79fe3512d192477fb402ac89/rskj-core/src/main/java/co/rsk/trie/TrieImpl.java#L623): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [TrieImpl.java#L628](https://github.com/rsksmart/rskj/blob/b4eff67aba79fb7a79fe3512d192477fb402ac89/rskj-core/src/main/java/co/rsk/trie/TrieImpl.java#L628): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [TrieImpl.java#L1056](https://github.com/rsksmart/rskj/blob/b4eff67aba79fb7a79fe3512d192477fb402ac89/rskj-core/src/main/java/co/rsk/trie/TrieImpl.java#L1056): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [TrieImpl.java#L1079](https://github.com/rsksmart/rskj/blob/b4eff67aba79fb7a79fe3512d192477fb402ac89/rskj-core/src/main/java/co/rsk/trie/TrieImpl.java#L1079): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![INFO][INFO] [TrieImpl.java#L60](https://github.com/rsksmart/rskj/blob/b4eff67aba79fb7a79fe3512d192477fb402ac89/rskj-core/src/main/java/co/rsk/trie/TrieImpl.java#L60): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)
1. ![INFO][INFO] [TrieImpl.java#L255](https://github.com/rsksmart/rskj/blob/b4eff67aba79fb7a79fe3512d192477fb402ac89/rskj-core/src/main/java/co/rsk/trie/TrieImpl.java#L255): Unconstrained method co.rsk.trie.TrieImpl.fromMessage(byte[], int, int, TrieStore) converts checked exception to unchecked [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3AEXS_EXCEPTION_SOFTENING_NO_CONSTRAINTS)
1. ![INFO][INFO] [TrieImpl.java#L728](https://github.com/rsksmart/rskj/blob/b4eff67aba79fb7a79fe3512d192477fb402ac89/rskj-core/src/main/java/co/rsk/trie/TrieImpl.java#L728): Unconstrained method co.rsk.trie.TrieImpl.deserialize(byte[]) converts checked exception to unchecked [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3AEXS_EXCEPTION_SOFTENING_NO_CONSTRAINTS)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by tinchou** [comment]
I think your tests are great, but they're not uncovering any bug. If you're changing `TrieImpl` because there's a bug, please add a failing test first.

Thanks!

File: rskj-core/src/test/java/co/rsk/trie/TrieImplDeleteTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
+package co.rsk.trie;
```
Comment (tinchou) [inline rskj-core/src/test/java/co/rsk/trie/TrieImplDeleteTest.java]: Missing License

File: rskj-core/src/test/java/co/rsk/trie/TrieImplDeleteTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+import org.ethereum.util.RLP;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static org.ethereum.crypto.HashUtil.sha3;
+import static org.ethereum.util.ByteUtil.EMPTY_BYTE_ARRAY;
+
+/**
+ * Created by ajlopez on 06/05/2017.
```
Comment (tinchou) [inline rskj-core/src/test/java/co/rsk/trie/TrieImplDeleteTest.java]: Please remove, you're not @ajlopez ;).

**Comment by donequis** [comment]
Tests were added and created by ajlopez.

The bug it fixes can be reproduced with the fooBar2.json test

**Comment by donequis** [comment]
There's no need to keep this pr here.

**Pull request closed without merging**